hw/ip/otbn/doc/_index.md

```diff
@@ -417,6 +417,21 @@ Additionally, instructions and data stored in the instruction and data memory, r
 
 Refer to the [Data Integrity Protection]({{<relref "#design-details-data-integrity-protection">}}) section for details of how the data integrity protections are implemented.
 
+## Secure Wipe
+
+OTBN provides a mechanism to securely wipe all state it stores, including the instruction memory.
+
+This wiping mechanism is split into three parts:
+- [Data memory secure wipe]({{<relref "#design-details-secure-wipe-dmem">}})
+- [Instruction memory secure wipe]({{<relref "#design-details-secure-wipe-imem">}})
+- [Internal state secure wipe]({{<relref "#design-details-secure-wipe-internal">}})
+
+The secure wipe mechanism can be triggered by host software through the bus interface using register {{< regref "SEC_WIPE">}}, and is automatically initiated in situations where OTBN has determined that the state it stores may have been compromised.
+
+The internal state secure wipe mechanism is triggered when executing the [`ECALL`]({{< relref "isa#ecall" >}}) instruction.
```

Comment:
This is effectively all one paragraph explaining the trigger mechanisms. Maybe make that clearer with text like this?

```suggestion
A secure wipe is performed automatically in certain situations, or can be requested manually by the host software.
A secure wipe of all state is performed when a fatal error is detected. 
(TODO: Specify lifecycle state dependency.)
A secure wipe of the internal state only is performed whenever an OTBN operation finishes.
Finally, host software can manually trigger a secure wipe operation by writing to the {{< regref "SEC_WIPE">}} register.
```

(Leave the TODO in there for now, it's something we need to figure out later.)

Reply:
Thinking further: we certainly want to do a internal secure wipe on a recoverable error (since that's identical to just terminating an OTBN operation early). We might even want to do a full secure wipe in that case. The list of errors which cause a recoverable error is at https://docs.opentitan.org/hw/ip/otbn/doc/index.html#Reg_err_bits and they could be both a programmer error, or a side-effect of a fault injection. 

The answer probably brings us back to the debuggability vs. security discussion.

@vrozic can you update the document to always include internal wipe for recoverable alerts, and leave a TODO for the potential full secure wipe? I could imagine the outcome being "full secure wipe if in PROD mode, otherwise internal wipe only."